File: rpn_calculator.rb
class RPNCalculator
	
	def evaluate(input)
		elements = input.split
		calculate(elements)
	end		

	def calculate(elements)
		if (elements.length > 1)
			for i in 0..(elements.length - 1)
				if (elements[i] == '+' || elements[i] == '-' || elements[i] == '/' || elements[i] == '*')
					result = eval(elements[i-2].to_s + elements[i].to_s + elements[i-1].to_s)					
					(elements[i-2] = result) && elements.delete_at(i-1) && elements.delete_at(i-1)
					calculate(elements)
				end				
			end
		end
		return elements[0].to_i
	end

end

calc = RPNCalculator.new
calc.evaluate('1 2 3 + -')
calc.evaluate('2 5 *')
calc.evaluate('50 20 -')
calc.evaluate('70 10 4 + 5 * -')
calc.evaluate('0')
calc.evaluate('1 1 +')
calc.evaluate('20 10 5 4 + * -')

File: rpn_calculator.js
// Array Remove - By John Resig (MIT Licensed)
Array.prototype.remove = function(from, to) {
  var rest = this.slice((to || from) + 1 || this.length);
  this.length = from < 0 ? this.length + from : from;
  return this.push.apply(this, rest);
};

//PostFix Calculator
function evaluate(input){
    var elem = input.split(' ');
    return calculate(elem);
}

function calculate(elem){
    if (elem.length > 1){
        for (var i=0; i< (elem.length); i++){
            if (elem[i] === "+" || elem[i] === "-" || elem[i] === "*"){
                elem[i-2] = eval(elem[i-2].toString() + elem[i].toString()+elem[i-1].toString());
                elem.remove(i-1) && elem.remove(i-1); 
                calculate(elem);
            }
        }
    }
    return Number(elem[0]);
}

console.log(evaluate("5 2 3 4 + * -"));
